project/profiles/capacity4more/modules/c4m/user/c4m_user_notifications/includes/c4m_user_notifications_handlers.inc

```diff
@@ -581,38 +588,99 @@ function c4m_user_notifications_content_recommended($nid) {
+ */
+function c4m_user_notifications_group_content_shared($source_node, $target_gid) {
+  _c4m_user_notifications_notify_on_content($source_node, 'share', $target_gid);
+}
+
+/**
+ * Email notification about existing content update at organic group.
+ */
+function c4m_user_notifications_existing_group_content_updated($node) {
+  if ($node->vid !== $node->original->vid) {
+    // New revision of content was published => notify content editors
+    // and group followers.
+    _c4m_user_notifications_notify_on_content_new_revision($node, 'content_editors');
+    _c4m_user_notifications_notify_on_content_new_revision($node, 'group_followers');
+  }
+  else {
+    // No revision update => => notify only group followers.
+    _c4m_user_notifications_notify_on_content($node, 'update');
+  }
 }
 
 /**
  * Helper function.
  *
- * Sends mail notification to group subscribers, when new content is added.
+ * Sends mail notification to group subscribers, when content is
+ * added, updated or shared.
  *
  * @param object $node
- *   Content node that was added.
- */
-function _c4m_user_notifications_new_content_published($node) {
+ *   Content node that was added / updated.
+ * @param string $action
+ *   Use 'insert' when new content is added.
+ *   Use 'update' when existing content is updated.
+ *   Use 'share' when content is shared.
+ * @param int $target_gid
+ *   Group ID for which notification is issued. If not passed, it's set to
+ *   gid of group node belong to.
+ */
+function _c4m_user_notifications_notify_on_content($node, $action, $target_gid = NULL) {
   global $user;
-  $message = message_create('c4m_new_content_published');
+
+  switch ($action) {
+    case 'insert':
+      $message_type = 'c4m_new_content_published';
+      $event_action = 'added';
+      break;
+
+    case 'share':
+      $message_type = 'c4m_new_content_published';
+      $event_action = 'shared';
+      break;
+
+    case 'update':
+      $message_type = 'c4m_content_updated';
+      break;
+
+    default:
+      return;
+  }
+
+  $message = message_create($message_type);
   // Define functions with which to resolve email template tokens.
-  c4m_message_load_tokens__c4m_new_content_published($message);
+  c4m_message_load_tokens__c4m_content_inserted_or_updated($message);
   // Resolve comment body field, in teaser mode, and assign value to it's token.
   $body_teaser = field_view_field('node', $node, 'c4m_body', 'teaser');
   $message->arguments['@node_teaser_body'] = strip_tags($body_teaser[0]['#markup']);
+  if (isset($event_action)) {
+    $message->arguments['@event_action'] = $event_action;
+  }
 
   $wrapper = entity_metadata_wrapper('message', $message);
   $wrapper->field_node->set($node->nid);
   $node_wrapper = entity_metadata_wrapper('node', $node);
-  $gid = $node_wrapper->{OG_AUDIENCE_FIELD}->value(array('identifier' => TRUE));
+  $gid = isset($target_gid) ? $target_gid : $node_wrapper->{OG_AUDIENCE_FIELD}->value(array('identifier' => TRUE));
```

Comment:
`$target_id` is always set since it is declared in the signature of the function, use `is_null()` or `empty()`